### DISCUSSION TITLE
django-admin startproject tutorial conflicts with pip module

### DISCUSSION THREAD

**Discussion by brunojustino** [encode/django-rest-framework#9517] [Potential Issue]
Trying to follow the: [Tutorial 1: Serialization]( https://www.django-rest-framework.org/tutorial/1-serialization/) I created the venv and installed the 3 packages as instructed. Then when I try to: `django-admin startproject tutorial` it says it conflicts with and existing Python module. Using other name for the project works ok. There is a tutorial module inside the site-packages probably bundled with one of the packages that I installed with pip but it's conflicting with the project name. Maybe it's a good a idea to check why is this module being installed or change the documentation to startproject with any other name.
![pythonresttutorialerror](https://github.com/user-attachments/assets/7ce1aa6b-5b9e-413d-99bd-22484f9cca09)

**Comment by browniebroke** [comment]
What's the content of the `tutorial` folder in site-packages? There is [no PyPI package](https://pypi.org/project/tutorial/) with this name so it's perhaps coming from another deps

**Comment by brunojustino** [comment]
So, I created a new project to test. I started a new virtual environment using vscode. In the print I'm sending it's the site-packages after running each step. 1 - vscode created the venv 2 - pip install django 3 - pip install djangorestframework 4 - it's the contents of the folder.
As you can see the tutorial module come bundled after the djangorestframework install.
![djangorestinstall](https://github.com/user-attachments/assets/6306ee3a-644a-46ae-8147-269bd7e1325f)

**Comment by browniebroke** [comment]
I can reproduce with the following steps:

```
python -m venv venv
source venv/bin/activate 
pip install djangorestframework
```

Here is the directory structure:

```
 ➜ tree venv/lib/python*/site-packages/tutorial                                  
venv/lib/python3.12/site-packages/tutorial
├── __init__.py
├── asgi.py
├── quickstart
│   ├── __init__.py
│   ├── admin.py
│   ├── apps.py
│   ├── migrations
│   │   └── __init__.py
│   ├── models.py
│   ├── serializers.py
│   ├── tests.py
│   └── views.py
├── settings.py
├── urls.py
└── wsgi.py
```

Here is the content of the `serializers.py` file:

```python
from django.contrib.auth.models import Group, User
from rest_framework import serializers


class UserSerializer(serializers.HyperlinkedModelSerializer):
    class Meta:
        model = User
        fields = ['url', 'username', 'email', 'groups']


class GroupSerializer(serializers.HyperlinkedModelSerializer):
    class Meta:
        model = Group
        fields = ['url', 'name']
```

I tried to rebuild the wheel and source distribution from the repo and it's not there, but when I download them from PyPI, they are included in the tarball & wheel, I guess a `tutorial/` folder may have been on disk when the release was built?

**Comment by browniebroke** [comment]
This sounds like a packaging bug: the content of the source distribution and wheel aren't matching what's in the repo. 

Is it worth publishing a new 3.15.2-post, same as 3.1.5.2 but without this extra content? It will require one of the maintainers to do it (I think @tomchristie @auvipy @peterthomassen - sorry for tagging).

As a side note, is there an appetite to automate the release process? Feels like there is room for human error at the moment, and PyPI supports [trusted publisher](https://docs.pypi.org/trusted-publishers/) which enable publishing from GH actions without having a long lived PyPI token.

**Reply by auvipy** [reply]
3.15.3 with one revert of supported python django versions. I can do that part

**Reply by peterthomassen** [reply]
I suppose you mean the changes suggested here: https://github.com/encode/django-rest-framework/issues/9450#issuecomment-2321314318

In this comment, @tomchristie is asking what other regression fixes should be included to close the 3.14 regression chapter. I've listed in the [next comment](https://github.com/encode/django-rest-framework/issues/9450#issuecomment-2340419747) the ones I'm aware of. I suggest we include fixes for those as well when we release 3.15.3.

If you agree with this, I can do the following:
- ask reporters for reproducible minimal examples / test
- bisect history to identify relevant commit
- see how to revert that commit (potentially with help from original approver)

What do you think?
